test/mocks/upstream/mocks.h

```diff
@@ -308,5 +308,11 @@ class MockClusterInfoFactory : public ClusterInfoFactory, Logger::Loggable<Logge
                     Event::Dispatcher& dispatcher, Runtime::RandomGenerator& random));
 };
 
+class MockRetryHostPredicate : public RetryHostPredicate {
+public:
```

Comment:
Please add explicit constructor/destructor and put definition in `.cc` for mock build performance reasons.

Reply:
Surprised this makes a difference. Does the implicit ctor/dtor definition in header add significant overhead?

Reply:
Surprising to me too, but see https://github.com/google/googletest/blob/master/googlemock/docs/CookBook.md#making-the-compilation-faster.

Reply:
Sure, any cleanups PRs most welcome.